lib/puppet/defaults.rb

```diff
@@ -582,14 +582,28 @@ def self.valid_file_checksum_types
       :type       => :duration,
       :desc       => "The minimum time to wait between checking for updates in
       configuration files.  This timeout determines how quickly Puppet checks whether
-      a file (such as manifests or templates) has changed on disk. #{AS_DURATION}",
+      a file (such as manifests or puppet.conf) has changed on disk. The default will
+      change in a future release to be 'unlimited', requiring a reload of the Puppet
+      service to pick up changes to its internal configuration. To reparse files
+      within an environment in Puppet Server please use the environment_cache endpoint",
+      :hook => proc do |val|
+        if ![0, '15s'].include?(val)
+          Puppet.deprecation_warning(<<-WARNING)
+            Fine grained control of filetimeouts is deprecated. In future
+            releases this value will only determine if file content is cached.
+
+            Valid values will be 0 (never cache) and 'unlimited' (always cache).
+            With a default of 'unlimited'.
```

Comment:
Seems a little odd that the default is not one of these values...

Reply:
I guess reading it again the current comment does imply that it will work that way in the future. I think this is fine.